homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,188 @@
+                        coordinator, dev.id, TemperatureMeasurement.Ambient
+                    )
+                )
+                device_registry = await dr.async_get_registry(hass)
+                device_registry.async_get_or_create(
+                    config_entry_id=entry.entry_id,
+                    identifiers={(DOMAIN, dev.id)},
+                    name="Meater Probe " + dev.id,
+                    manufacturer="Apption Labs",
+                    model="Meater Probe",
+                )
+                hass.data[DOMAIN]["entities"][dev.id] = None
+
+        async_add_entities(entities)
+
+        return devices
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="meater_api",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+    )
+
+    def null_callback():
+        return
+
+    # Add a subscriber to the coordinator that doesn't actually do anything, just so that it still updates when all probes are switched off
+    coordinator.async_add_listener(null_callback)
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+
+class MeaterProbeTemperature(Entity):
+    """Meater Temperature Sensor Entity."""
+
+    def __init__(self, coordinator, device_id, temperature_reading_type):
+        """Initialise the sensor."""
+        self.coordinator = coordinator
+        self.device_id = device_id
+        self.temperature_reading_type = temperature_reading_type
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Meater Probe " + self.temperature_reading_type.name
+
+    @property
+    def state(self):
+        """Return the temperature of the probe."""
+        # First find the right probe in the collection
+        device = None
+
+        for dev in self.coordinator.data:
+            if dev.id == self.device_id:
+                device = dev
+
+        if device is None:
+            return None
+
+        if TemperatureMeasurement.Internal == self.temperature_reading_type:
+            return device.internal_temperature
+
+        # Not an internal temperature, must be ambient
+        return device.ambient_temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+
+        # Meater API always return temperature in Celsius
+        return TEMP_CELSIUS
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return "temperature"
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.device_id)
+            },
+            "name": self.name,
```

Comment:
```suggestion
            "name": "Meater Probe",
```
Using `self.name` results in the device being called `Meater Probe Ambient`. I'm not entirely sure if this would be an issue if multiple probes are added, I don't think so but I also only have 1 probe to test with.

Reply:
Since there are two sensors, isn't it useful to keep the "Ambient" label so that it is known which is measuring the ambient air temperature?

Reply:
There are 2 sensor entities, one for ambient and one for internal temperature, but only 1 device that the 2 entities belong to. This only changes the name of the device to `Meater Probe`, and the 2 entities would still be named appropriately. Hope that makes sense.

Reply:
Yes, the Meater Probe Ambient was incorrectly being used for the device name in this case

Reply:
that does make sense. thanks for the clarification